client/app/components/Modal/index.jsx

```diff
@@ -83,90 +112,75 @@ export class Modal extends React.Component<Props, State> {
+        {displayModalHeader()}
+        {displayModalBody()}
       </div>
     </div>
   );
 
-  handleClick = () => {
-    const { modalHasFocus } = this.state;
-    if (modalHasFocus) return;
-    this.toggleOpen();
-  };
-
-  handleKeyPress = (event: SyntheticKeyboardEvent<HTMLDivElement>) => {
-    if (event.key !== 'Escape') return;
-    this.toggleOpen();
-  };
-
-  setModalHasFocus = (modalHasFocus: boolean) => {
-    this.setState({ modalHasFocus });
-  };
-
-  toggleOpen = () => {
-    const { open } = this.state;
-    const { openListener } = this.props;
-    const body = ((document.body: any): HTMLBodyElement);
-    if (!open) {
-      body.classList.add('bodyModalOpen');
-    } else {
-      body.classList.remove('bodyModalOpen');
-    }
-    if (!open && openListener) {
-      openListener();
+  const resolveComponent = (component: string) => {
+    /** Really only returns Avatar right now but more could be added if needed */
+    switch (component) {
+      case 'Avatar':
+      default:
+        return Avatar;
     }
-    this.setState({ open: !open });
   };
 
-  resolveElement = () => {
-    const { element, elementId } = this.props;
+  const resolveElement = () => {
     let renderComponent;
     if (element && element.component) {
-      const { component, props } = element;
-      renderComponent = React.createElement(this.resolveComponent(component), {
-        ...props,
+      const { component, props: elementProps } = element;
+      renderComponent = React.createElement(resolveComponent(component), {
+        ...elementProps,
       });
     }
     if (element) {
       return (
         <div
           id={elementId}
           className={`modalElement ${css.modalElement}`}
-          onClick={this.toggleOpen}
-          onKeyDown={this.toggleOpen}
+          onClick={toggleOpen}
+          onKeyDown={toggleOpen}
           role="button"
           tabIndex={0}
         >
-          {renderComponent || this.displayContent(element)}
+          {renderComponent || displayContent(element)}
         </div>
       );
     }
     return null;
   };
 
-  resolveComponent = (component: string) => {
-    /** Really only returns Avatar right now but more could be added if needed */
-    switch (component) {
-      case 'Avatar':
-      default:
-        return Avatar;
-    }
-  };
+  return (
+    <>
+      {resolveElement()}
+      {open ? displayModalBox() : null}
+    </>
+  );
+};
 
-  render() {
-    const { open } = this.state;
-    const renderElement = this.resolveElement();
-    return (
-      <div>
-        {renderElement}
-        {open ? this.displayModalBox() : null}
-      </div>
-    );
-  }
-}
+// There's a [bug](https://github.com/shakacode/react_on_rails/issues/1198) with React on Rails,
+// so we'll need to do this in order to render multiple components with hooks on the same page.
+export default ({
+  element,
+  elementId,
+  title,
+  openListener,
+  open,
+  body,
+}: Props) => (
+  <Modal
```

Comment:
So in `QuickCreate` we render this modal this way:
```
<Modal body={body} title={label} open={open} key={modalKey} />
```

Previously we didn't need to explicitly use `key` in this component, but I would look into debugging this since the failing test is related to the this value.